examples/go/stream_benchmarker/benchmark/main.go

```diff
@@ -0,0 +1,261 @@
+		SatelliteID:     *satelliteID,
+		APIKey:          *apiKey,
+		Endpoint:        *endpoint,
+	}
+
+	var streamChannel = make(chan *stellarstation.Telemetry)
+
+	ss, err := benchmark.OpenSatelliteStream(satelliteStreamOptions, streamChannel)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	ticker := time.NewTicker(time.Second * time.Duration(*interval))
+	defer ticker.Stop()
+
+	fmt.Println("Listening for messages")
+	var sessionSummary sessionSummary
+	var md metricsData
+	var storeMostRecentTimeLastByteReceived time.Time = time.Time{}
+	var gotFirstMessage = false
+	var done = false
+	for !done {
+		select {
+		case message := <-streamChannel:
+			if gotFirstMessage == false {
+				fmt.Println("Receiving messages")
+				printHeader()
+				gotFirstMessage = true
+				firstTime, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+				md.initialTime = firstTime
+				sessionSummary.firstByteTime = firstTime
+			}
+			timeFirstByteReceived, _ := ptypes.Timestamp(message.TimeFirstByteReceived)
+			timeLastByteReceived, _ := ptypes.Timestamp(message.TimeLastByteReceived)
+			numBytesInMessage := len(message.Data)
+
+			metric := metric{
+				timeFirstByteReceived: timeFirstByteReceived,
+				timeLastByteReceived:  timeLastByteReceived,
+				dataSize:              numBytesInMessage,
+				metricTime:            time.Now().UTC(),
+			}
+			md.metrics = append(md.metrics, metric)
+			storeMostRecentTimeLastByteReceived = timeLastByteReceived
+			md.mostRecentTimeLastByteReceived = storeMostRecentTimeLastByteReceived
+		case timeOfReporting := <-ticker.C:
+			if gotFirstMessage {
+				md.timeOfReporting = timeOfReporting
+				outputDetails := md.compileDetails()
+				sessionSummary.totalBytes += outputDetails.totalDataSize
+				sessionSummary.totalMbps += outputDetails.mbps
+				sessionSummary.totalAverageFirstByteLatency += outputDetails.averageFirstByteLatency
+				sessionSummary.totalAverageLastByteLatency += outputDetails.averageLastByteLatency
+				sessionSummary.numIntervals++
+				sessionSummary.lastByteTime = md.mostRecentTimeLastByteReceived
+				sessionSummary.totalOutOfOrderData += outputDetails.outOfOrderData
+				if (*exitOnNoDataInSeconds > 0) && (outputDetails.totalDataSize == 0) {
+					secondsSinceFirstByte := time.Now().UTC().Sub(sessionSummary.firstByteTime)
+					if secondsSinceFirstByte > time.Duration(*exitOnNoDataInSeconds) {
+						done = true
+					}
+				}
+				if !done {
+					printDetails(outputDetails)
+					md = metricsData{}
+					md.initialTime = timeOfReporting
+					md.mostRecentTimeLastByteReceived = storeMostRecentTimeLastByteReceived
+				}
+			}
+		case <-signalChan:
+			fmt.Println("\nReceived interrupt, stopping benchmark")
+			done = true
+		}
+	}
+	if *printSessionSummary {
+		if sessionSummary.numIntervals > 0 {
+			printSummary(&sessionSummary)
+		}
+	}
+	ss.Close()
```

Comment:
Move this to a defer right after you check for the error from creating this object.  That way it gets called even if there is a panic.

Reply:
I think you might be looking at a previous commit, because there is a defer ss.Close() right after checking for the error from creating SatelliteStream object.